File: app/src/main/java/com/parassidhu/popularmovies/viewmodels/MovieViewModelFactory.java
package com.parassidhu.popularmovies.viewmodels;

import android.app.Application;
import android.arch.lifecycle.ViewModel;
import android.arch.lifecycle.ViewModelProvider;
import android.support.annotation.NonNull;

import com.parassidhu.popularmovies.database.MovieDatabase;

public class MovieViewModelFactory extends ViewModelProvider.NewInstanceFactory {
    private final Application application;

    public MovieViewModelFactory(Application application) {
        this.application = application;
    }

    public <T extends ViewModel> T create(@NonNull Class<T> modelClass){
        return (T) new MovieViewModel(application);
    }
}

File: app/src/main/java/com/parassidhu/popularmovies/viewmodels/MovieViewModelFactory.kt
package com.parassidhu.popularmovies.viewmodels

import android.app.Application
import android.arch.lifecycle.ViewModel
import android.arch.lifecycle.ViewModelProvider

import com.parassidhu.popularmovies.database.MovieDatabase

class MovieViewModelFactory(private val application: Application) : ViewModelProvider.NewInstanceFactory() {

    override fun <T : ViewModel> create(modelClass: Class<T>): T {
        return MovieViewModel(application) as T
    }
}
